var/spack/repos/builtin/packages/py-pyomo/package.py

```diff
@@ -80,25 +85,29 @@ class PyPyomo(PythonPackage):
     depends_on("py-pybind11", when="@6.1:+tests", type=("run"))
 
     # when docs is requested
-    depends_on("py-sphinx@4:", when="@6.1:+docs", type=("run"))
+    depends_on("py-sphinx@5:", when="@6.1:+docs", type=("run"))
```

Comment:
I'm not actually sure if this is accurate. pyomo 6.1 requires sphinx@3:, not @5:. You should go back through the history and see when it changed. Same for other dependencies.